### PR TITLE
sk - seed database with some images in new deployments.

### LINKED ISSUE #4: Image Seeds
#### As a developer I want at least 20 images to be created when I setup the database.

__Story__:

Working on this application has been a lot of fun, right? You are fast
becoming a git expert, and an expert at working with a deployment
pipeline. Speaking of development and deployment pipelines, is it not pretty
annoying that everytime a new deployment is created it starts out with an empty
database? It seems like everytime a review application is created we need to
manually add a handful of images in order to manually test the
application. Would it not be great if each deployment was created with a
handful of existing images?

__Acceptance criteria__:
- [ ] After running `rake db:setup` at least 20 unique images are shown in the
  homepage.
- [ ] When a new app is deployed on heroku (e.g., every review app), that app
  automatically has at least 20 unique images on its homepage.

__Dependencies__:
- Image Index

### PR DESCRIPTION
**Pull request by skapfer**
Closes #4 

seed the database with some example images for new deployments.

FIXME: how to test this? db:seed populates the RAILS_ENV=development even when CircleCI deploys for "rake test". Does it even make sense to test this? Should the test work on the production DB?

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [ ] I have replaced `X` above with the associated issue number.
- [ ] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [ ] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

**Comment by spjsschl** [comment]
very good point about testing; you do not need to test your test data or (mostly) your test code; we test (mostly) only production code.

File: db/seeds.rb [hunk 1]
```diff
@@ -5,3 +5,39 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
```
Comment (spjsschl) [inline db/seeds.rb]: these comments can be removed; they are not needed anymore

File: db/seeds.rb [hunk 2]
```diff
@@ -5,3 +5,39 @@
...
+  'https://upload.wikimedia.org/wikipedia/commons/thumb/e/ec/Tabby_Pfaffengrund.JPG/1024px-Tabby_Pfaffengrund.JPG',
+  'https://upload.wikimedia.org/wikipedia/commons/thumb/2/20/Penny_snuggles.JPG/1024px-Penny_snuggles.JPG',
+  'https://upload.wikimedia.org/wikipedia/commons/6/60/Neighbours_Siamese.jpg',
+  'https://upload.wikimedia.org/wikipedia/commons/thumb/6/68/Orange_tabby_cat_sitting_on_fallen_leaves-Hisashi-01A.jpg/800px-Orange_tabby_cat_sitting_on_fallen_leaves-Hisashi-01A.jpg',
+  'https://upload.wikimedia.org/wikipedia/commons/b/b9/Cats_eye.jpg',
+  'https://upload.wikimedia.org/wikipedia/commons/7/70/Gatos_cats_7_cropped.jpg'
+]
+
+cat_pictures.each do |url|
+  Imageurl.create!(url: url)
```
Comment (spjsschl) [inline db/seeds.rb]: you can pass an array to create!

**Comment by skapfer** [comment]
> very good point about testing; you do not need to test your test data or (mostly) your test code; we test (mostly) only production code.

The thing is, this is not test data. It's the initial data that should go into a new deployment so the app is not empty when it is first seen by the user.

I guess this should be tested in some way, but probably at the deployment level?

File: db/seeds.rb [hunk 3]
```diff
@@ -1,7 +1,37 @@
...
+  'https://upload.wikimedia.org/wikipedia/commons/thumb/e/ec/Tabby_Pfaffengrund.JPG/1024px-Tabby_Pfaffengrund.JPG',
+  'https://upload.wikimedia.org/wikipedia/commons/thumb/2/20/Penny_snuggles.JPG/1024px-Penny_snuggles.JPG',
+  'https://upload.wikimedia.org/wikipedia/commons/6/60/Neighbours_Siamese.jpg',
+  'https://upload.wikimedia.org/wikipedia/commons/thumb/6/68/Orange_tabby_cat_sitting_on_fallen_leaves-Hisashi-01A.jpg/800px-Orange_tabby_cat_sitting_on_fallen_leaves-Hisashi-01A.jpg',
+  'https://upload.wikimedia.org/wikipedia/commons/b/b9/Cats_eye.jpg',
+  'https://upload.wikimedia.org/wikipedia/commons/7/70/Gatos_cats_7_cropped.jpg'
+]
+
+cat_picture_urls = cat_pictures.map { |x| { url: x } }
+Imageurl.create! cat_picture_urls
```
Comment (spjsschl) [inline db/seeds.rb]: the seatle bracing style is OK in ruby but can learn to tricky problems. unless it is a DSL context it is usually better to avoid it

**Pull request closed without merging**